fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/synthetic/JoinedRecordPlanner.java

```diff
@@ -261,9 +271,138 @@ private static QueryComponent buildCondition(@Nonnull KeyExpression expression,
                 default:
                     throw new RecordCoreException("unsupported fan type in join key expression: " + expression);
             }
+        } else if (expression instanceof InvertibleFunctionKeyExpression) {
+            final InvertibleFunctionKeyExpression function = (InvertibleFunctionKeyExpression)expression;
+            final Comparisons.Comparison inverted = invertFunctionComparison(function, comparison);
+            return buildCondition(function.getArguments(), inverted);
         } else {
             throw new RecordCoreException("unsupported join key expression: " + expression);
         }
     }
 
+    @Nonnull
+    private static Comparisons.Comparison invertFunctionComparison(@Nonnull InvertibleFunctionKeyExpression function, @Nonnull Comparisons.Comparison comparison) {
+        // Limit function comparisons (for now) to unary functions
+        if (function.getColumnSize() != 1) {
+            throw new RecordCoreException("invertible functions in joins must return a single element");
+        }
+        if (function.getMinArguments() != 1 || function.getMaxArguments() != 1) {
+            throw new RecordCoreException("invertible functions in joins must operate on a single element");
+        }
+        return new InvertFunctionComparisonWithParameter(function, comparison);
+    }
+
+    private static class InvertFunctionComparisonWithParameter implements Comparisons.Comparison {
```

Comment:
Is it worth trying to have this outside of just the joined record planner? I guess the main question is how you'd ever get one of these comparisons otherwise, absent improvements to `QueryKeyExpressionWithComparison`.

But it would allow for a unit test that planning a suitable inverted comparison gives an index scan with bounds computed by inverting a parameter value at execution time.